tests/mbt/model/libraries/extraSpells.qnt

```diff
@@ -139,6 +139,71 @@ module extraSpells {
         __set.fold(List(), (__l, __e) => __l.append(__e))
     }
 
+  /// The type of orderings between comparable things
+  // Follows https://hackage.haskell.org/package/base-4.19.0.0/docs/Data-Ord.html#t:Ordering
+  // and we think there are likely benefits to using 3 constant values rather than the more
+  // common integer range in Apalache.
+  type Ordering =
+    | EQ
+    | LT
+    | GT
+
+  /// Comparison of integers
+  pure def intCompare(__a: int, __b:int): Ordering = {
+    if (__a < __b)
+      { LT }
+    else if (__a > __b)
+      { GT }
+    else
+      { EQ }
+  }
+
+  /// Assuming `__l` is sorted according to `__cmp`, returns a list with the element `__x`
+  /// insterted in order.
+  ///
+  /// If `__l` is not sorted, `__x` will be inserted after the first element less than
+  /// or equal to it.
+  ///
+  /// - @param __l a sorted list
+  /// - @param __x an element to be inserted
+  /// - @param __cmp an operator defining an `Ordering` of the elemnts of type `a`
+  /// - @returns a sorted list that includes `__x`
+  pure def sortedListInsert(__l: List[a], __x: a, __cmp: (a, a) => Ordering): List[a] = {
+    // We need to track whether __x has been inserted, and the accumulator for the new list
+    val __init = { is_inserted: false, acc: List() }
+
+    val __result = __l.foldl(__init, (__state, __y) =>
+      if (__state.is_inserted)
+        { ...__state, acc: __state.acc.append(__y) }
+      else
+        match __cmp(__x, __y) {
+          | GT => { ...__state, acc: __state.acc.append(__y) }
+          | _  => { is_inserted: true, acc: __state.acc.append(__x).append(__y)  }
+        })
+
+    if (not(__result.is_inserted))
+      // If __x was not inserted, it was GT than every other element, so it goes at the end
+      __result.acc.append(__x)
+    else
+      __result.acc
+  }
+
+  run sortedListInsertTest = all {
+    assert(List().sortedListInsert(3, intCompare) == List(3)),
+    assert(List(1,2,4).sortedListInsert(3, intCompare) == List(1,2,3,4)),
+    assert(List(4,1,2).sortedListInsert(3, intCompare) == List(3,4,1,2)),
```

Comment:
This seems a peculiar test because it does not satisfy the precondition of `sortedListInsert`, namely that `List(4, 1, 2)` is sorted. Might make sense to replace with a test that has a non-empty list and we append something at the its tail.